service/templates/java/ControllerTest.java

```diff
@@ -4,19 +4,24 @@
 import io.micronaut.http.HttpStatus;
 import io.micronaut.http.client.RxHttpClient;
 import io.micronaut.runtime.server.EmbeddedServer;
-import org.junit.Test;
+import io.micronaut.test.annotation.MicronautTest;
 
+import org.junit.jupiter.api.Test;
 
-import static org.junit.Assert.assertEquals;
+import javax.inject.Inject;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@MicronautTest
 public class ${className}Test {
 
+    @Inject
+    EmbeddedServer embeddedServer;
+
     @Test
     public void testIndex() throws Exception {
-        try(EmbeddedServer server = ApplicationContext.run(EmbeddedServer.class)) {
-            try(RxHttpClient client = server.getApplicationContext().createBean(RxHttpClient.class, server.getURL())) {
-                assertEquals(HttpStatus.OK, client.toBlocking().exchange("/${propertyName}").status());
-            }
+        try(RxHttpClient client = embeddedServer.getApplicationContext().createBean(RxHttpClient.class, embeddedServer.getURL())) {
+            assertEquals(HttpStatus.OK, client.toBlocking().exchange("/${propertyName}").status());
```

Comment:
This can be simplified to 

```java
@Inject @Client("/")
RxHttpClient client
```

Without the need for the try block and injecting the embedded server etc.